apply_DDT.py

```diff
@@ -156,393 +160,511 @@ def main():
 
     set_mpl_fontsize(18, 22, 26)
 
-    # grab training features for specific input model
-    global training_features
-    training_features = read_training_features(model_file)
-
-    # add ddt necessary variables in addition to the bdt input features
-    all_features = training_features + ['pt', 'mt', 'rho', 'rt'] # rho is an important variable for applying the decorrelation
-
-    # Grab the bkg data
-    bkg_cols = [Columns.load(f) for f in glob.glob(bkg_files)]
-    X, pT, mT, rho, bkg_weight = bdt_ddt_inputs(bkg_cols, lumi, all_features)
-
-    # _____________________________________________
-    # Open the trained models and get the scores
-
-    bkg_score = {}
-    bkg_score = calc_bdt_scores(X, model_file=model_file)
+    #--------------------------------------------------------------------------
+    # For the cut-based search ------------------------------------------------
+    #--------------------------------------------------------------------------
 
-    # _____________________________________________
-    # Calculate bkg efficiencies for the DDT
+    if ana_type == 'cut-based' :
```

Comment:
these two sections (cut-based and BDT-based) can be reunified by making a dictionary of the specific values and operations that vary:
```python
cut_specific = {'cut-based': {}, 'BDT-based': {}}
```
the entries should be:
1. a function to get the features (including running the BDT in the BDT-based case), which when called, separates out the cut variable from the other features (so it can be used generically in operations like `ecfm2b1[ecfm2b1>0.09]` for either case)
2. a list of cut values (i.e. just [0.09] for cut-based)
3. plot names, axis names

then you just call `cut_specific[ana_type]["key"]` whenever you need a cut-specific value at any point during the process

Reply:
Very tricky to implement, but I think that Yi-Mu and I got it. We just finished testing it and are pleased with the results.